src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import styles from '@/components/Checkbox/Checkbox.module.scss'
+import { Text } from '@/components/Text/Text'
+import csx from 'classnames'
+
+interface CheckboxProps {
+  name: string
+  label: string
+  disabled?: boolean
+}
+
+const Checkbox: React.FC<CheckboxProps> = ({
+  name,
+  label,
+  disabled = false,
+}) => {
+  const classes = csx(
+    styles.checkbox,
+    disabled ? styles.checkbox__disabled : ''
+  )
+
+  const disabledProp = disabled ? { disabled: true } : {}
+
+  return (
+    <label className={classes} htmlFor={name}>
+      <input name={name} type="checkbox" id={name} {...disabledProp} />
```

Comment:
- `onClick` handler prop is missing
- input should be controlled by props so add a `checked` prop

```suggestion
    <label className={classes} htmlFor={name} onClick={onClick}>
      <input name={name} type="checkbox" id={name} disabled={disabled} checked={checked}/>
```